Rdmp.Core/DataExport/Checks/SelectedDataSetsChecker.cs

```diff
@@ -167,7 +168,10 @@ public void Check(ICheckNotifier notifier)
                         catch (Exception e)
                         {
                             if (server.GetQuerySyntaxHelper().IsTimeout(e))
-                                notifier.OnCheckPerformed(new CheckEventArgs("Failed to read rows after " + timeout + "s", CheckResult.Warning,e));
+                            {
+                                if (UserSettings.WarnOnTimeoutOnExtractionChecks)
```

Comment:
I don't like adding if statements all over the place, can we add an extension method to ICheckNotifier that takes a code or enum? then we can build a bit more maintainability around this stuff?